### PR TITLE
Fix some bugs found by Sonar

### PR DESCRIPTION
**Pull request by projectkk2glider**
I am trying to fix some problems detected by Sonar.

Do not merge (yet)!

### CODE REVIEW

File: radio/src/lua/widgets.cpp [hunk 1]
```diff
@@ -113,7 +113,6 @@ ZoneOption * createOptionsArray(int reference, uint8_t maxOptions)
             }
             else if (option->type == ZoneOption::Source ||
                      option->type == ZoneOption::TextSize ||
-                     option->type == ZoneOption::Source ||
```
Comment (bsongis) [inline radio/src/lua/widgets.cpp:116]: interesting :)

File: radio/src/lua/widgets.cpp [hunk 2]
```diff
@@ -243,7 +242,7 @@ void luaLoadThemeCallback()
     LuaTheme * theme = new LuaTheme(name, options);
     theme->loadFunction = loadFunction;
     theme->drawBackgroundFunction = drawBackgroundFunction;
-    theme->drawTopbarBackgroundFunction = drawTopbarBackgroundFunction;
+    theme->drawTopbarBackgroundFunction = drawTopbarBackgroundFunction;   //NOSONAR
```
Comment (bsongis) [inline radio/src/lua/widgets.cpp]: can we write it like this `// NOSONAR`?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## OpenTX 2.2 Branch
 
 [![Travis build Status](https://travis-ci.org/opentx/opentx.svg?branch=2.2)](https://travis-ci.org/opentx/opentx)
-[![Quality Gate](https://sonarcloud.io/api/badges/gate?key=OpenTX)](https://sonarcloud.io/dashboard?id=OpenTX)
```
Comment (bsongis) [inline README.md:4]: Yes I saw that, we are not in the green!

**Comment by bsongis** [comment]
Can I merge it now? Or you have more errors fixed?

**Comment by projectkk2glider** [comment]
Merge freely. I just wonder why Sonar did not post anything about this branch.

**Comment by bsongis** [comment]
you lucky, sonar is quick at spams ...

**Pull request merged**